homeassistant/components/media_player/__init__.py

```diff
@@ -463,7 +432,8 @@ def supported_features(self):
 
     def turn_on(self):
         """Turn the media player on."""
-        raise NotImplementedError()
+        if self.supported_features & SUPPORT_TURN_ON:
```

Comment:
What happens if `supported_features` isn't declared or doesn't contain `SUPPORT_TURN_ON`.

Reply:
My understanding of how `media_player` works!

Reply:
:) Right, it's a [property](https://github.com/home-assistant/home-assistant/blob/01e1fc4aeea04b9ef87f98b31ff510f606248c17/homeassistant/components/media_player/__init__.py#L429).